cpp/src/arrow/type.h

```diff
@@ -1582,13 +1583,23 @@ class ARROW_EXPORT FieldRef {
 // ----------------------------------------------------------------------
 // Schema
 
+enum class Endianness { LITTLE, BIG };
+#if ARROW_LITTLE_ENDIAN
+#define NATIVE_ENDIANNESS Endianness::LITTLE
+#else
+#define NATIVE_ENDIANNESS Endianness::BIG
```

Comment:
Could you hide this from public API?
It doesn't have `ARROW_` prefix.

Reply:
sure, done